packages/toolpad-app/src/toolpad/hooks/useUnsavedChangesConfirm.ts

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+
+type OnCloseType = (...args: unknown[]) => void | Promise<void>;
+
+interface UseUnsavedChangesConfirmInput {
+  hasUnsavedChanges: boolean;
+  onClose: OnCloseType;
+}
+
+interface UseUnsavedChangesConfirmPayload {
+  handleCloseWithUnsavedChanges: OnCloseType;
+}
+
+export default function useUnsavedChangesConfirm({
+  hasUnsavedChanges,
+  onClose,
+}: UseUnsavedChangesConfirmInput): UseUnsavedChangesConfirmPayload {
+  const handleCloseWithUnsavedChanges = React.useCallback(() => {
+    if (hasUnsavedChanges) {
+      // eslint-disable-next-line no-alert
+      const hasConfirmed = window.confirm(
+        'You have unsaved changes. Are you sure you want to navigate away?\nAll changes will be discarded.',
```

Comment:
```suggestion
        'You have unsaved changes. Are you sure you want to navigate away? All changes will be discarded when you press "Ok".',
```

Maybe the newline looks a bit odd?

![Screenshot 2023-02-20 at 17 23 19](https://user-images.githubusercontent.com/2109932/220158054-4790079a-4c2a-4f25-b7ee-e252f7b933dc.png)